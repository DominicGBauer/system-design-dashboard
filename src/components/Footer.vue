<template>
  <footer>
    <h4>AFRICAN INSTITUTE OF FINANCIAL MARKETS AND RISK MANAGEMENT</h4>
    <p>Faculty of Commerce</p>
    <p>University of Cape Town</p>
    <p>
      Â© 2021 Copyright:
      <a href="http://www.aifmrm.uct.ac.za/"> AIFMRM</a>
    </p>
  </footer>
</template>

<style lang="scss" scoped>
footer {
  background-color: #2c3e50;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 2rem;

  p {
    margin: 6px;
  }

  a {
    color: green;
  }
}
</style>
