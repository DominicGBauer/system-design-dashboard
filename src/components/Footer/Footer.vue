<template>
  <div class="footer-container">
    <footer>
      <h4>AFRICAN INSTITUTE OF FINANCIAL MARKETS AND RISK MANAGEMENT</h4>
      <p>Faculty of Commerce</p>
      <p>University of Cape Town</p>
      <p>
        Â© 2021 Copyright:
        <a href="http://www.aifmrm.uct.ac.za/"> AIFMRM</a>
      </p>
    </footer>
  </div>
</template>

<style lang="scss" scoped>
.footer-container {
  margin-top: auto;

  footer {
    margin-top: 5%;
    background-color: #2c3e50;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 2rem;

    p {
      margin: 6px;
    }

    a {
      color: green;
    }
  }
}
</style>
