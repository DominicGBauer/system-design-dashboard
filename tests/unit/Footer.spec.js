import { shallowMount } from '@vue/test-utils'
import Footer from '@/components/Footer.vue'

describe('Footer.vue', () => {
  it('renders the Footer text', () => {
    const wrapper = shallowMount(Footer, {})
    expect(wrapper.text()).toMatch(
      'AFRICAN INSTITUTE OF FINANCIAL MARKETS AND RISK MANAGEMENTFaculty of CommerceUniversity of Cape Town Â© 2021 Copyright: AIFMRM',
    )
  })
})
