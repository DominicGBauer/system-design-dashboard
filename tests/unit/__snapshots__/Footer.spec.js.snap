// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue renders the Footer and it looks the same 1`] = `
<footer>
  <h4>
    AFRICAN INSTITUTE OF FINANCIAL MARKETS AND RISK MANAGEMENT
  </h4>
  <p>
    Faculty of Commerce
  </p>
  <p>
    University of Cape Town
  </p>
  <p>
     Â© 2021 Copyright: 
    <a
      href="http://www.aifmrm.uct.ac.za/"
    >
       AIFMRM
    </a>
  </p>
</footer>
`;
